grails-app/services/org/pih/warehouse/data/LoadDataService.groovy

```diff
@@ -200,52 +209,120 @@ class LoadDataService {
+        )
+
+        // 2a. If there are available items:
+        //   - If inventory snapshot is turned on for load demo data:
+        //       - then create an inventory snapshot transaction with entries made from the current stock calculated in 1st point
+        // 2b. If there are no available items:
+        //   - then no snapshot is being saved
+        if (availableItems.size() && isInventoryBaselineEnabled) {
+            inventoryImportProductInventoryTransactionService.createInventoryBaselineTransactionForGivenStock(
+                    targetWarehouse,
+                    null,
+                    availableItems.values(),
+                    inventoryBaselineTransactionDate
+            )
+        }
+
+        // 3. Create an adjustment (credit) transaction and take the data imported
+        Transaction adjustmentTransaction = createAdjustmentTransaction(adjustmentTransactionDate, targetWarehouse.inventory)
+
+        rows.forEach { Map<String, String> attr ->
+            Product product = productMap[attr["Product code"]]
+            Location binLocation = locationMap[attr["Bin location"]]
 
-        csvReader.eachLine { Map<String, String> attr ->
-            Product product = Product.findByProductCode(attr["Product code"]);
             if (!product) {
                 throw new IllegalArgumentException("Product not found: " + attr["Product code"])
             }
 
-            Location binLocation = Location.findByName(attr["Bin location"])
-
             if (!binLocation) {
                 throw new IllegalArgumentException("Location not found: " + attr["Bin location"])
             }
 
-            TransactionEntry transactionEntry = new TransactionEntry();
-            transactionEntry.quantity = attr["Physical QOH"] as Integer
-            transactionEntry.comments = attr["Comment"]
-
-            def expirationDate = attr["Expiration date"]
-
-            InventoryItem inventoryItem = inventoryService.findAndUpdateOrCreateInventoryItem(
+            TransactionEntry transactionEntry = createAdjustmentTransactionEntry(
+                    attr,
                     product,
-                    attr["Lot number"],
-                    expirationDate == ""
-                            ? null
-                            : dateFormat.parse(expirationDate)
+                    binLocation,
+                    availableItems
             )
 
-            transactionEntry.inventoryItem = inventoryItem
-            transactionEntry.binLocation = binLocation
-
-            transaction.addToTransactionEntries(transactionEntry)
-        };
+            if (transactionEntry) {
+                adjustmentTransaction.addToTransactionEntries(transactionEntry)
+            }
+        }
 
-        transaction.forceRefresh = Boolean.TRUE
-        transaction.save(flush: true, failOnError: true)
+        adjustmentTransaction.forceRefresh = Boolean.TRUE
+        adjustmentTransaction.save(flush: true, failOnError: true)
```

Comment:
We briefly discussed this in my PR, but I switched the logic in inventory import to be:
```
if (!adjustmentTransaction.save()) {
    throw new ValidationException("Invalid transaction", adjustmentTransaction.errors)
}
```
Any reason we can't do the same here? Why is the flush and force refresh need?

Reply:
Also, you'll want to only create the transaction if adjustmentTransaction.transactionEntries == true. There's no need to create the transaction if it'd have no entries